modules/webgl/src/adapter/resources/webgl-render-pipeline.ts

```diff
@@ -175,6 +175,12 @@ export class WEBGLRenderPipeline extends RenderPipeline {
   }
 
   setUniforms(uniforms: Record<string, UniformValue>) {
+    Object.keys(uniforms).filter(k => !isUniform(uniforms[k])).forEach(key => {
```

Comment:
`splitUniformsAndBindings()` could help here as well... if not, type out variable names as above.